### DISCUSSION TITLE
Utility methods with multiple nodes

### DISCUSSION THREAD

**Discussion by hossein-zare** [socketio/socket.io#4395] [Q&A] [answered]
Hi,

I'm using some of the utility methods such as `fetchSockets` and `disconnectSockets`.
Are they available for multiple nodes with the Redis Adapter?

```js
io.in('room-name').disconnectSockets();
```

By the way, Are socket objects read-only on other servers if we use `fetchSockets`?

**Comment by darrachequesne** [comment, marked as answer]
Hi! Yes, the Redis adapter is compatible with those methods.

> By the way, Are socket objects read-only on other servers if we use fetchSockets?

On other servers, you'll get a object which exposes a subset of the methods of the Socket class:

```js
const sockets = await io.fetchSockets();

for (const socket of sockets) {
  console.log(socket.id);
  console.log(socket.handshake);
  console.log(socket.rooms);
  console.log(socket.data);
  socket.emit(/* ... */);
  socket.join(/* ... */);
  socket.leave(/* ... */);
  socket.disconnect(/* ... */);
}
```

**Reply by hossein-zare** [reply]
Cool!

> On other servers, you'll get a object which exposes a subset of the methods of the Socket class:
> ...

Is there any way to do something like `socket.data.age = 25` for remote objects?

**Reply by darrachequesne** [reply]
In that case, you will have to use the `serverSideEmit()` method:

```js
// on server A
io.serverSideEmit("setAge", id, age);

// on server B

io.on("setAge", (id, age) => {
  if (io.sockets.sockets.has(id)) {
    const socket = io.sockets.sockets.get(id);

    socket.data.age = age;
  }
});
```

Documentation: https://socket.io/docs/v4/server-instance/#serversideemit
